src/main/java/core/basesyntax/serviceimpl/DataConverterImpl.java

```
@@ -0,0 +1,25 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (int i = 1; i < inputReport.size(); i++) {
+            String[] lineArray = inputReport.get(i).split(",");
+            String code = lineArray[0].trim();
+            String fruit = lineArray[1].trim();
+            int quantity = Integer.parseInt(lineArray[2].trim());
```

The code calls `Integer.parseInt(lineArray[2].trim())` without handling parsing errors or negative quantities. The checklist instructs: "Don't ignore exceptions in try-catch blocks ... Let's rethrow a RuntimeException with an informative message and exception object." . Also the task requires: "Validate quantity is an integer and not negative; throw on invalid values." . Wrap parsing with a try/catch to rethrow a RuntimeException that includes the line content/index, and check `quantity >= 0` before continuing.